go/test/endtoend/throttler/util.go

```diff
@@ -534,11 +528,11 @@ func WaitForValidData(t *testing.T, tablet *cluster.Vttablet, timeout time.Durat
 
 	for {
 		checkResp, checkErr := http.Get(checkURL)
-		if checkErr != nil {
+		if checkErr == nil {
```

Comment:
We don't want to have a `require.NoError(t, checkErr)` here before the defer, getting rid of the conditional check? Same below.

Reply:
Not in this case, because this is a wait-for function, and I'm OK ignoring the error as long as the function eventually succeeds within the given timeout. So I don't care that there could be multiple errors, and I don't want to report the test as failed.